tests/test_cc.py

```diff
@@ -0,0 +1,142 @@
+        """Stderr is printed if ca-certificates install fails."""
+        self.ca_certificates.unlink()
+        self.make_fake_binary('apt-get', command='echo failed >&2; false')
+        process = self.script('install-cc', 'user:pass')
+        self.assertEqual(1, process.returncode)
+        self.assertIn('failed', process.stderr)
+
+    def test_install_cc_missing_token(self):
+        """The token must be specified when using enable-fips."""
+        process = self.script('install-cc')
+        self.assertEqual(3, process.returncode)
+        self.assertIn(
+            'Invalid token, it must be in the form "user:password"',
+            process.stderr)
+
+    def test_install_cc_invalid_token_format(self):
+        """The CC token must be specified as "user:password"."""
+        process = self.script('install-cc', 'foo-bar')
+        self.assertEqual(3, process.returncode)
+        self.assertIn(
+            'Invalid token, it must be in the form "user:password"',
+            process.stderr)
+
+    def test_install_cc_invalid_token(self):
+        """If token is invalid, an error is returned."""
+        message = (
+            'E: Failed to fetch https://esm.ubuntu.com/'
+            '  401  Unauthorized [IP: 1.2.3.4]')
+        self.make_fake_binary(
+            'apt-helper', command='echo "{}"; exit 1'.format(message))
+        process = self.script('install-cc', 'user:pass')
+        self.assertEqual(3, process.returncode)
+        self.assertIn('Checking token... ERROR', process.stdout)
+        self.assertIn('Invalid token', process.stderr)
+
+    def test_install_cc_error_checking_token(self):
+        """If token check fails, an error is returned."""
+        message = (
+            'E: Failed to fetch https://esm.ubuntu.com/'
+            '  404  Not Found [IP: 1.2.3.4]')
+        self.make_fake_binary(
+            'apt-helper', command='echo "{}"; exit 1'.format(message))
+        process = self.script('install-cc', 'user:pass')
+        self.assertEqual(3, process.returncode)
+        self.assertIn('Checking token... ERROR', process.stdout)
+        self.assertIn(
+            'Failed checking token (404  Not Found [IP: 1.2.3.4])',
+            process.stderr)
+
+    def test_install_cc_skip_token_check_no_helper(self):
+        """If apt-helper is not found, the token check is skipped."""
+        self.apt_helper.unlink()
+        process = self.script('install-cc', 'user:pass')
+        self.assertEqual(0, process.returncode)
+        self.assertIn('Checking token... SKIPPED', process.stdout)
+
+    def test_install_cc_only_supported_on_xenial(self):
+        """The install-cc option fails if not on Xenial."""
+        self.SERIES = 'zesty'
+        process = self.script('install-cc', 'user:pass')
+        self.assertEqual(4, process.returncode)
+        self.assertIn(
+            'Canonical Common Criteria is not supported on zesty',
+            process.stderr)
```

Comment:
Please also add a test for the (un)supported architectures. Looks like arm is unsupported, so I think it's ok to just add that negative test.

Reply:
Added 2 tests for arm64 and i686 for unsupported architectures.